src/js/jsx/Panel.jsx

```diff
@@ -26,107 +26,37 @@ define(function (require, exports, module) {
-                recentFiles: recentFiles,
-                activeDocumentID: activeDocumentID,
-                documentIDs: documentIDs,
-                mountedDocumentIDs: mountedDocumentIDs
-            };
-        },
-
-        shouldComponentUpdate: function (nextProps, nextState) {
-            // Don't re-render if we're just going temporarily inactive so that
-            // the UI doesn't blink unnecessarily.
-            if (this.props.active && !nextProps.active) {
-                return false;
-            }
+    var Properties = React.createClass({
 
-            // Don't re-render until either the active document or recent files
-            // are initialized.
-            if (!nextState.activeDocumentInitialized ||
-                (!this.state.document && !nextState.recentFilesInitialized)) {
+        shouldComponentUpdate: function (nextProps) {
+            // The document is inactive
+            if (!nextProps.current) {
                 return false;
             }
 
-            return this.state.activeDocumentID !== nextState.activeDocumentID ||
-                this.state.activeDocumentInitialized !== nextState.activeDocumentInitialized ||
-                this.state.recentFilesInitialized !== nextState.recentFilesInitialized ||
-                (nextState.documentIDs.size === 0 && !Immutable.is(this.state.recentFiles, nextState.recentFiles));
+            return true;
         },
-
+        
         render: function () {
-            var documentIDs = this.state.documentIDs;
-
-            if (this.state.activeDocumentInitialized && documentIDs.size > 0) {
-                var activeDocumentID = this.state.activeDocumentID,
-                    documentProperties = this.state.mountedDocumentIDs.map(function (documentID) {
-                        var current = documentID === activeDocumentID,
-                            className = classnames({
-                                "panel__element": true,
-                                "panel__element__active": current
-                            });
-
-                        return (
-                            <div className={className} key={documentID}>
-                                <Properties
-                                    documentID={documentID}
-                                    current={current} />
-                            </div>
-                        );
-                    }, this);
-                return (
-                    <div className="panel">
-                        {documentProperties}
+            var className = classnames({
+                    "panel": true,
+                    "panel__visible": this.props.visible
+                });
+
+            return (
+                <div className={className}>
+                    <div
+                        className="panel__hide"
+                        onClick={this.props.onVisibilityToggle}>
+                        ▸
```

Comment:
In my experience, there is a high probability that this character will not render correctly on Windows. Also, I'm probably going to start a design argument about this particular UI signifier, but not for this PR.